### DISCUSSION TITLE
Alpine focus noscroll problems

### DISCUSSION THREAD

**Discussion by yungifez** [alpinejs/alpine#4317] [1. Help] [answered]
I wasn't sure if to label this as a bug, so I decided to post it here

Imagine I have 2 elements in a page, say a dropdown and a modal, they both have noscroll modifiers, and they both release focus when clicked anywhere outside. 

When the dropdown is open, it traps and causes a noscroll, but when the modal is opened, it sets the overflow back to scroll

Then clicking outside the modal sets the overflow to hidden, essentially freezing the scroll of the page

Here is a scraped together codepen 

https://codepen.io/yungifez/pen/wvLzwyE?editors=1100

And here is where I noticed this problem

https://x-aui.com/docs/0.x/components/dropdown-menu

**Comment by ekwoka** [comment, marked as answer]
You need to have focused things that focus other things close before opening the other thing, or do a proper cascading close where the inner closes and then the outer closes.

Since each of the focuses is designed to set the overflow back to whatever it was before it opened, when handled in isolation.

**Comment by yungifez** [comment]
Sorry, this was my own skill issue

**Comment by iongroup-github** [comment]
Can you please tell me how you fix this? The explanation above is very confusing.Thank you.

**Reply by yungifez** [reply]
You can close set an outside click to close whatever is open or unfocused what ever is focused

And then instead of showing or focus immediately, do this on nextTick
